## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 433065`

Logs: https://github.com/lorenzleutgeb/nixpkgs-review-gha/actions/runs/16907744632


---
### `x86_64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>radicle-node</li>
  </ul>
</details>


---
### `aarch64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>radicle-node</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>radicle-node</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>radicle-node</summary>
<pre>   |
19 | use rustix::fs::MemfdFlags;
   |     ^^^^^^^^^^^^^^^^^^^^^^ no `MemfdFlags` in `fs`

error[E0432]: unresolved import `rustix::fs::SealFlags`
  --> /nix/var/nix/builds/nix-build-radicle-node-1.3.0.drv-5437-3824494466/radicle-node-1.3.0-vendor/systemd-journal-logger-2.2.2/src/client.rs:20:5
   |
20 | use rustix::fs::SealFlags;
   |     ^^^^^^^^^^^^^^^^^^^^^ no `SealFlags` in `fs`

error[E0599]: no associated item named `NOSIGNAL` found for struct `SendFlags` in the current scope
  --> /nix/var/nix/builds/nix-build-radicle-node-1.3.0.drv-5437-3824494466/radicle-node-1.3.0-vendor/systemd-journal-logger-2.2.2/src/client.rs:95:24
   |
95 |             SendFlags::NOSIGNAL,
   |                        ^^^^^^^^ associated item not found in `SendFlags`

Some errors have detailed explanations: E0432, E0599.
For more information about an error, try `rustc --explain E0432`.
error: could not compile `systemd-journal-logger` (lib) due to 5 previous errors
warning: build failed, waiting for other jobs to finish...</pre>
</details>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>radicle-node</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>radicle-node</summary>
<pre>   |
19 | use rustix::fs::MemfdFlags;
   |     ^^^^^^^^^^^^^^^^^^^^^^ no `MemfdFlags` in `fs`

error[E0432]: unresolved import `rustix::fs::SealFlags`
  --> /nix/var/nix/builds/nix-build-radicle-node-1.3.0.drv-3417-1666944793/radicle-node-1.3.0-vendor/systemd-journal-logger-2.2.2/src/client.rs:20:5
   |
20 | use rustix::fs::SealFlags;
   |     ^^^^^^^^^^^^^^^^^^^^^ no `SealFlags` in `fs`

error[E0599]: no associated item named `NOSIGNAL` found for struct `SendFlags` in the current scope
  --> /nix/var/nix/builds/nix-build-radicle-node-1.3.0.drv-3417-1666944793/radicle-node-1.3.0-vendor/systemd-journal-logger-2.2.2/src/client.rs:95:24
   |
95 |             SendFlags::NOSIGNAL,
   |                        ^^^^^^^^ associated item not found in `SendFlags`

Some errors have detailed explanations: E0432, E0599.
For more information about an error, try `rustc --explain E0432`.
error: could not compile `systemd-journal-logger` (lib) due to 5 previous errors
warning: build failed, waiting for other jobs to finish...</pre>
</details>
</details>

--- a/pkgs/by-name/ra/radicle-node/package.nix
+++ b/pkgs/by-name/ra/radicle-node/package.nix
@@ -19,13 +19,13 @@
 }:
 rustPlatform.buildRustPackage rec {
   pname = "radicle-node";
-  version = "1.2.1";
+  version = "1.3.0";
   env.RADICLE_VERSION = version;
 
   src = fetchgit {
     url = "https://seed.radicle.xyz/z3gqcJUoA1n9HaHKufZs5FCSGazv5.git";
     rev = "refs/namespaces/z6MkireRatUThvd3qzfKht1S44wpm4FEWSSa4PRMTSQZ3voM/refs/tags/v${version}";
-    hash = "sha256-pqYV3n/aKNbEDEp8v4oQUMMlsSiJZq/nh5gFP4KpZbM=";
+    hash = "sha256-0gK+fM/YGGpxlcR1HQixbLK0/sv+HH29h6ajEP2w2pI=";
     leaveDotGit = true;
     postFetch = ''
       git -C $out rev-parse HEAD > $out/.git_head
@@ -34,7 +34,7 @@ rustPlatform.buildRustPackage rec {
     '';
   };
 
-  cargoHash = "sha256-T457lXW0M2eO9R+8XyobUFVj4NOiXVSvtDztp1i0PS4=";
+  cargoHash = "sha256-qLRFZXbVbsgMyXiljsb8lOBCDZKa17LcxWuPaUYSG70=";
 
   nativeBuildInputs = [
     asciidoctor
@@ -47,6 +47,14 @@ rustPlatform.buildRustPackage rec {
     export GIT_HEAD=$(<$src/.git_head)
     export SOURCE_DATE_EPOCH=$(<$src/.git_time)
   '';
+
+  cargoBuildFlags = [
+    "--package=radicle-node"
+    "--package=radicle-cli"
+    "--package=radicle-remote-helper"
+  ];
+
+  cargoTestFlags = cargoBuildFlags;
 
   # tests regularly time out on aarch64
   doCheck = stdenv.hostPlatform.isLinux && stdenv.hostPlatform.isx86;

